projects/dotcms-field-elements/src/components/dot-form/dot-form.e2e.ts

```diff
@@ -0,0 +1,106 @@
+        {
+            fieldType: 'Text',
+            defaultValue: 'defaultValue',
+            hint: 'hint2',
+            name: 'field2',
+            required: true,
+            value: 'value2',
+            variable: 'field2'
+        }
+    ];
+
+    beforeEach(async () => {
+        page = await newE2EPage();
+        await page.setContent(
+            `<dot-form submit-label="Save" reset-label="Reset"></dot-form>`
+        );
+        element = await page.find('dot-form');
+        element.setProperty('fields', fields);
+        await page.waitForChanges();
+        const txtFields = await element.findAll('dot-textfield');
+        txtFields.forEach((field, index) => {
+            field.triggerEvent('valueChanges', {
+                bubbles: true,
+                cancelable: false,
+                detail: {
+                    name: fields[index].name,
+                    value: fields[index].value
+                }
+            });
+        });
+        await page.waitForChanges();
+        element.getProperty('_formValues').then((data) => { formStatus = data; });
+    });
+
+    it('should renders', async () => {
+        // tslint:disable-next-line:max-line-length
+        const tagsRenderExpected = `<form><dot-textfield class="hydrated"><label for="field1">field1</label><input name="field1" type="text" required=""><span class="dot-textfield__hint">hint1</span></dot-textfield><dot-textfield class="hydrated"><label for="field2">field2</label><input name="field2" type="text" required=""><span class="dot-textfield__hint">hint2</span></dot-textfield><button type="submit">Save</button><button type="button">Reset</button></form>`;
+        expect(element.innerHTML).toBe(tagsRenderExpected);
+    });
+
+    it('should send "formSubmit" event', async () => {
+        const expectedSubmit = {};
+        const spy = await page.spyOnEvent('formSubmit');
+        const saveBtn = await element.find('button[type="submit"]');
+
+        fields.forEach((field) => {
+            expectedSubmit[field.name] = field.value;
+        });
+
+        saveBtn.click();
+        await page.waitForChanges();
+        expect(spy).toHaveReceivedEventDetail(expectedSubmit);
+    });
+
+    it('should listen for valueChanges', async () => {
+        const textField = await page.find('dot-textfield');
+        const newValue = {
+            name: 'field1',
+            value: 'test2'
+        };
+
+        textField.triggerEvent('valueChanges', {
+            bubbles: true,
+            cancelable: false,
+            detail: newValue
+        });
+
+        formStatus = {...formStatus, field1: 'test2' };
+
+        await page.waitForChanges();
+        element.getProperty('_formValues').then((data) => {
+            expect(data).toEqual(formStatus);
+        });
+    });
+
+    it('should reset event', async () => {
+        const resetBtn = await element.find('button[type="button"]');
+        const expectedStatus = Object.assign({}, formStatus);
```

Comment:
What is the value of `formStatus` at this point? is this value being change by other tests? if so, we need to make sure all the tests start with their own state, modify value of object in different tests leads to hard to track issues in the tests.

Reply:
based on this:
https://github.com/dotCMS/core-web/blob/7b377e18fc57ae03a7f0da8392151c95d2dbdf7f/projects/dotcms-field-elements/src/components/dot-form/dot-form.e2e.ts#L8

the value is `{ field1: value1, field2: value2 }` --> that is being set on `beforeEach` of every test